File: glean-core/ios/Glean/Metrics/MemoryUnit.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import Foundation

/// Enumeration of different resolutions supported by
/// the `MemoryDistributionMetricType`.
///
/// These use the power-of-2 values of these units, that is, Kilobyte is pedantically a Kibibyte.
public enum MemoryUnit: Int32 {
    /// 1
    case byte = 0

    /// 2^10
    case kilobyte = 1

    /// 2^20
    case megabyte

    /// 2^30
    case gigabyte
}

File: glean-core/android/src/main/java/mozilla/telemetry/glean/private/MemoryUnit.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package mozilla.telemetry.glean.private

/**
 * Enumeration of different resolutions supported by the MemoryDistribution metric type.
 *
 * These use the power-of-2 values of these units, that is, Kilobyte is pedantically a Kibibyte.
 */
enum class MemoryUnit {
    /**
     * Byte: 1 byte
     */
    Byte,

    /**
     * Kilobyte: 2^10 bytes
     */
    Kilobyte,

    /**
     * Megabyte: 2^20 bytes
     */
    Megabyte,

    /**
     * Gigabyte: 2^30 bytes
     */
    Gigabyte
}
